flink-core/src/main/java/org/apache/flink/api/connector/source/SourceReader.java

```diff
@@ -32,7 +33,8 @@
  * @param <SplitT> The type of the the source splits.
  */
 @PublicEvolving
-public interface SourceReader<T, SplitT extends SourceSplit> extends AutoCloseable {
+public interface SourceReader<T, SplitT extends SourceSplit>
+		extends AutoCloseable, CheckpointListener {
```

Comment:
Optional: We could add a default method override for `notifyCheckpointComplete()` to indicate that this is not a "core method" but somewhat optional for some cases.
That would also save us overriding it in each test class (with an empty method).

I am fine either way, would leave this up to you.